### DISCUSSION TITLE
How to use import.meta.url in webpack.ts

### DISCUSSION THREAD

**Discussion by tommyboylab** [webpack/webpack#17854] [Q&A]
I'm wondering how to use the new convention `import.meta.url` in webpack.ts instead of the depreciated __dirname.

I tried adding a flag to the import property (`<reference types="webpack/module" />`) and a flag under the module section but both fail to work.

error I get is this: ` The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'es2022', 'esnext', 'system', 'node16', or 'nodenext'.`

But from docs I need ts-node to use commonjs and so cannot change the module option.

https://webpack.js.org/configuration/configuration-languages/#typescript

**Comment by alexander-akait** [comment]
Hello, do you still need a help?
